prow/plugins/label/label_test.go

```diff
@@ -0,0 +1,176 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package label
+
+import (
+	"fmt"
+	"testing"
+
+	"k8s.io/kubernetes/pkg/util/sets"
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/github/fakegithub"
+)
+
+const (
+	fakeRepoOrg  = "fakeOrg"
+	fakeRepoName = "fakeName"
+	orgMember    = "Alice"
+	nonOrgMember = "Bob"
+	prNumber     = 1
+)
+
+var areaLabels = []string{"area/api", "area/infra"}
+var priorityLabels = []string{"priority/important", "priority/critical"}
+
+func formatLabels(labels sets.String) sets.String {
+	r := sets.NewString()
+	for l := range labels {
+		r.Insert(fmt.Sprintf("%s/%s#%d:%s", fakeRepoOrg, fakeRepoName, prNumber, l))
+	}
+	return r
+}
+
+func getFakeRepo(commentBody, commenter string, initial_labels []string) (*fakegithub.FakeClient, github.IssueCommentEvent) {
+	fc := &fakegithub.FakeClient{
+		IssueComments: make(map[int][]github.IssueComment),
+		OrgMembers:    []string{orgMember},
+	}
+	startingLabels := []github.Label{}
+	for _, l := range initial_labels {
+		startingLabels = append(startingLabels, github.Label{Name: l})
+	}
+
+	ice := github.IssueCommentEvent{
+		Repo: github.Repo{
+			Owner: github.User{Name: fakeRepoOrg},
+			Name:  fakeRepoName,
+		},
+		Comment: github.IssueComment{
+			Body: commentBody,
+			User: github.User{Login: commenter},
+		},
+		Issue: github.Issue{
+			User:        github.User{Login: "a"},
+			Number:      prNumber,
+			PullRequest: &struct{}{},
+			Labels:      startingLabels,
+		},
+	}
+	return fc, ice
+}
+
+func TestLabel(t *testing.T) {
+	// "a" is the author, "a", "r1", and "r2" are reviewers.
+	var testcases = []struct {
+		name              string
+		body              string
+		commenter         string
+		expectedNewLabels sets.String
+		removedLabels     []string
+		initialLabels     []string
+		isPr              bool
+	}{
+		{
+			name:              "Irrelevant comment",
+			body:              "irrelelvant",
+			expectedNewLabels: sets.NewString(),
+			commenter:         orgMember,
+		},
+		{
+			name:              "Empty Area",
+			body:              "/area",
+			expectedNewLabels: formatLabels(sets.NewString()),
```

Comment:
You're test doesn't really detect failures :-/ I wish we knew what happens here.